libmcount/dynamic.c

```diff
@@ -73,6 +74,39 @@ static struct mcount_orig_insn *lookup_code(struct rb_root *root,
 	return iter;
 }
 
+static inline bool frozen(struct code_page *cp)
+{
+	return cp->frozen;
+}
+
+static inline int release(struct code_page *cp)
+{
+	int res = 0;
+
+	res = mprotect(cp->page, CODE_CHUNK, PROT_READ|PROT_EXEC|PROT_WRITE);
+	if (res == 0)
+		cp->frozen = false;
+	else
+		pr_dbg("release failed as unexpected.");
```

Comment:
I think it's dangerous to make the code page writable after it contains executable code actually.  I think it's better to allocate a new separate codepage for dlopen().

Reply:
code changed as your advice.